packages/api-bindings/src/apollo/createProfileTypePolicy.ts

```diff
@@ -1,9 +1,47 @@
 import { ReactiveVar } from '@apollo/client';
+import { FollowPolicy, FollowPolicyType } from '@lens-protocol/domain/use-cases/profile';
 import { WalletData } from '@lens-protocol/domain/use-cases/wallets';
+import { Amount, ChainType, erc20, invariant } from '@lens-protocol/shared-kernel';
 
-import { ProfileAttributes, Profile, ProfileAttributeReader } from '../graphql';
+import {
+  FollowModule,
+  getFollowPolicyTypeFromProfileFieldsFragment,
+  isFeeFollowModule,
+  Profile,
+  ProfileAttributeReader,
+  ProfileAttributes,
+} from '../graphql';
 import { TypePolicy } from './TypePolicy';
 
+function resolveFollowPolicy({
+  followModule,
+}: {
+  followModule: FollowModule | null;
+}): FollowPolicy {
+  const followPolicyType = getFollowPolicyTypeFromProfileFieldsFragment(followModule);
+
+  if (followPolicyType === FollowPolicyType.CHARGE) {
+    invariant(isFeeFollowModule(followModule), 'Profile must have the fee follow module');
```

Comment:
Is this invariant needed?

If we were to check the `followModule.__typename` the discriminated union should be narrow down so this invariant and maybe the new `isFeeFollowModule` (if not used anywhere else) are not needed.

Maybe we are using the `getFollowPolicyTypeFromProfileFieldsFragment` too early. It seems is getting in the way.